EasyReflectometry/sample/assemblies/base_assembly.py

```diff
@@ -38,3 +53,109 @@ def __repr__(self) -> str:
         :rtype: str
         """
         return yaml.dump(self._dict_repr, sort_keys=False)
+
+    @property
+    def top_layer(self) -> Layer:
+        """
+        :return: The top layer
+        """
+        return self.layers[0]
+
+    @top_layer.setter
+    def top_layer(self, layer: Layer) -> None:
+        """
+        Setter for the top layer
+        """
+        self.layers[0] = layer
+
+    @property
+    def bottom_layer(self) -> Layer:
+        """
+        :return: The bottom layer
+        """
+        return self.layers[-1]
```

Comment:
Should be safe for an assembly with just one layer.  One might then discuss if the concept of a top and bottom layer makes sense in that case

Reply:
Code now checks if there a more than one layer